extensions/amp-story-auto-ads/0.1/story-ad-page.js

```diff
@@ -265,14 +266,26 @@ export class StoryAdPage {
           this.localizationService_
         ) || uiMetadata[A4AVarNames.CTA_TYPE];
 
-      // Store the cta-type as an accesible var for any further pings.
-      this.analytics_.then((analytics) =>
+      this.analytics_.then((analytics) => {
+        // Store the cta-type as an accesible var for any further pings.
         analytics.setVar(
           this.index_, // adIndex
           AnalyticsVars.CTA_TYPE,
           uiMetadata[A4AVarNames.CTA_TYPE]
-        )
-      );
+        );
+
+        // Set meta tag based variables.
+        const metaTags = (this.adDoc_ ?? this.adElement_).querySelectorAll(
```

Comment:
This logic is very similar to https://github.com/ampproject/amphtml/blob/main/extensions/amp-story-auto-ads/0.1/story-ad-ui.js#L75

would it make more since to move it there?

Reply:
Updated.